lib/client/api_test.go

```diff
@@ -499,6 +501,51 @@ func TestApplyProxySettings(t *testing.T) {
 	}
 }
 
+type testAgent struct {
+	agent.Agent
+}
+
+type mockSigner struct {
+	ssh.Signer
+}
+
+func (s *mockSigner) PublicKey() ssh.PublicKey {
+	return &ssh.Certificate{
+		ValidPrincipals: []string{"VALID"},
+	}
+}
+
+// Signers implements agent.Agent
+func (*testAgent) Signers() ([]ssh.Signer, error) {
+	return []ssh.Signer{&mockSigner{}}, nil
+}
+
+func TestGetProxySSHPrincipal(t *testing.T) {
+	client, err := NewClient(&Config{
+		Username:         "xyz",
+		HostLogin:        "xyz",
+		WebProxyAddr:     "localhost",
+		SkipLocalAuth:    true,
+		UseKeyPrincipals: true, // causes VALID to be returned, as key was used
+		Agent:            &testAgent{},
+		AuthMethods:      []ssh.AuthMethod{ssh.Password("xyz") /* placeholder authmethod */},
+	})
```

Comment:
Maybe:

```suggestion
	cfg := &Config{
		Username:         "xyz",
		HostLogin:        "xyz",
		WebProxyAddr:     "localhost",
		SkipLocalAuth:    true,
		UseKeyPrincipals: true, // causes VALID to be returned, as key was used
		Agent:            &testAgent{},
		AuthMethods:      []ssh.AuthMethod{ssh.Password("xyz") /* placeholder authmethod */},
	}

	client, err := NewClient(cfg)
```

so we can just set `cfg.UseKeyPrincipals = false` and reuse it below? Makes it clear that `UseKeyPrincipals` is the only config toggle we care about.